test/metabase/pulse/render/common_test.clj

```diff
@@ -29,29 +31,48 @@
         (is (= "$12,345.54" (fmt {::mb.viz/currency "USD"}))))
       (testing "Other currencies"
         (is (= "AED12,345.54" (fmt {::mb.viz/currency "AED"})))
-        (is (= "₡12,345.54" (fmt {::mb.viz/currency "CRC"})))
+        (is (= "Af12,346" (fmt {::mb.viz/currency "AFN"})))
+        (is (= "₡12,346" (fmt {::mb.viz/currency "CRC"})))
+        (is (= "ZK12,346" (fmt {::mb.viz/currency "ZMK"})))
         (is (= "12,345.54 Cape Verdean escudos"
-               (fmt {::mb.viz/currency "CVE"
+               (fmt {::mb.viz/currency       "CVE"
                      ::mb.viz/currency-style "name"}))))
       (testing "Understands name, code, and symbol"
         (doseq [[style expected] [["name" "12,345.54 Czech Republic korunas"]
                                   ["symbol" "Kč12,345.54"]
                                   ["code" "CZK 12,345.54"]]]
-          (is (= expected (fmt {::mb.viz/currency "CZK"
+          (is (= expected (fmt {::mb.viz/currency       "CZK"
                                 ::mb.viz/currency-style style}))
               style))))
     (testing "scientific notation"
       (is (= "1.23E4" (fmt {::mb.viz/number-style "scientific"})))
-      (is (= "1.2346E4" (fmt {::mb.viz/number-style "scientific"
-                              ::mb.viz/decimals 4})))
+      (is (= "1.23E4" (fmt {::mb.viz/number-style "scientific"
+                            ::mb.viz/decimals     2})))
+      (is (= "1.235E4" (fmt {::mb.viz/number-style "scientific"
+                             ::mb.viz/decimals     3})))
+      (is (= "1.23455E4" (fmt {::mb.viz/number-style "scientific"
+                               ::mb.viz/decimals     5})))
       (is (= "1E4" (fmt {::mb.viz/number-style "scientific"
-                         ::mb.viz/decimals 0}))))
+                         ::mb.viz/decimals     0}))))
     (testing "Percentage"
       (is (= "1,234,554.32%" (fmt {::mb.viz/number-style "percent"})))
       (is (= "1.234.554,3200%"
-             (fmt {::mb.viz/number-style "percent"
-                   ::mb.viz/decimals 4
-                   ::mb.viz/number-separators ",."}))))
+             (fmt {::mb.viz/number-style      "percent"
+                   ::mb.viz/decimals          4
+                   ::mb.viz/number-separators ",."})))
+      (is (= "10%" (format 0.1 {::mb.viz/number-style "percent"})))
+      (is (= "1%" (format 0.01 {::mb.viz/number-style "percent"})))
+      (is (= "0.00001%" (format 0.0000001 {::mb.viz/number-style "percent"}))))
```

Comment:
leave a comment in the code that handles this natural formatting of numbers when no choices are made (0.0000001 remains un-rounded to 0, 40.00 -> 40, etc)